lib/SILOptimizer/Mandatory/TFLowerGraph.cpp

```diff
@@ -201,7 +201,21 @@ namespace {
 }
 
 namespace {
-struct TFGraphLowering : public SILInstructionVisitor<TFGraphLowering> {
+/// Status in the process of graph lowering. This is used for
+/// error propagation and handling in the TFGraphLowering class.
+enum class GLStatus {
+  // No error occurs.
+  Succeeded,
+
+  // SIL Instruction not handled in the SILInstructionVisitor
+  // implementation.
+  UnhandledSILInstr,
```

Comment:
Why have classifications?  I'd just make this be "Success" and "Error", and emit the diagnostic at the point of the failure.  The only purpose for this enum is to be propagated, there isn't anything at the top level that should switch on it.

Reply:
Having the simple success/error classifications is a good start. A possible use case of having more classifications (more generally, more info in the status object) is to log/alert on different error types differently. e.g. if someone ever runs the compiler in a cloud service, internal errors (any errors that are not caused by end users) could trigger special alerts that say page a developer / SRE.

Since this is mostly speculative, starting simple would make sense.